docs/docs/generating-nlu-data.mdx

```diff
@@ -3,3 +3,110 @@ id: generating-nlu-data
 sidebar_label: Generating NLU Data
 title: Generating NLU Data
 ---
+
+NLU (Natural Language Understanding) is the part of Rasa Open Source that performs intent classification, entity extraction, and response retrieval.
+
+NLU will take in a sentence such as "I am looking for a French restaurant in the center of town" and return structured data like:
+
+```json
+{
+  "intent" : "search_restaurant",
+  "entities" : {
+    "cuisine" : "French",
+    "location" : "center"
+  }
+}
+```
+
+Building NLU models is hard, and building ones that are production-ready is even harder. Here are some tips for designing your NLU training data and pipeline to get the most out of your bot.
+
+
+## Conversation-Driven Development for NLU
+
+Conversation-Driven Development (CDD) means letting real user conversations guide your development. For building a great NLU model, this means:
+
+
+### Gather real data (vs. generating synthetic data)
+When it comes to building out NLU training data, developers often rely on text generation tools to quickly increase the number of training examples. While this approach can be time-saving, it makes NLU models prone to overfit data that is not representative of things a real user would say.
+
+
+To avoid such a problem, it is always a good idea to collect as much real user data as possible to use as training data. Even though your bot will make mistakes initially, this process of training & evaluating on user data will enable your model to generalize much more effectively in real-world scenarios.
+
+### Share with test users early (vs. only trying it out yourself)
+
+In order to gather real data, you’re going to need real user messages. This means sharing your bot with test users outside the development team early on. See [these guidelines](./cdd) for more details.
+
+
+## Avoiding Intent Confusion
+
+A common mistake developers make is creating separate intents when entities can be used as the differentiating factor.
+An example of this is when designing intents centered around the user providing information. A simple solution would be to create separate intents depending on what information is provided, e.g. inform_name, inform_address, inform_email, when all of these intents should actually be grouped as a single inform intent with several possible entity values.
+
+
+Extracted entity values can be used to affect the conversation flow using categorical slots. For example:
+
+TODO: replace with new story format
+```story
+- inform
+    - slot{pet_type: "dog"}
+    - utter_dogs_are_cool
+
+- inform
+   - slot{pet_type: "cat"}
+   - utter_cats_are_cool_too
+```
+
+## Extracting Entities
```

Comment:
For this section I think we can move a lot of info from the nlu training data page (improving entity extraction and intent classification), but I think that's probably something to take care of after this alpha, its already on my to-do-list